### ISSUE TITLE
Inspec 3.6.6 fails to upload to Chef Automate

### ISSUE THREAD

**Issue by Kiranjith83** [inspec/inspec#3841]
### 🎛 Description

**Choose one:** is this a 🐛 bug report 
 Uploading compliance controls fails with inspec version 3.6.6. But works well with other version 3.2.6.

```
# inspec compliance upload centos-ssh-baseline --overwrite
WARN: Unresolved specs during Gem::Specification.reset:
      bundler (>= 1.10)
WARN: Clearing out unresolved specs.
Please report a bug if this causes problems.
Dependencies for profile centos-ssh-baseline successfully vendored to /root/centos-ssh-baseline/vendor
Traceback (most recent call last):
	13: from /bin/inspec:306:in `<main>'
	12: from /bin/inspec:306:in `load'
	11: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.6.6/bin/inspec:12:in `<top (required)>'
	10: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/base.rb:466:in `start'
	 9: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	 8: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	 7: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	 6: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:238:in `block in subcommand'
	 5: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:115:in `invoke'
	 4: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor.rb:387:in `dispatch'
	 3: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/invocation.rb:126:in `invoke_command'
	 2: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/thor-0.20.3/lib/thor/command.rb:27:in `run'
	 1: from /opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.6.6/lib/plugins/inspec-compliance/lib/inspec-compliance/cli.rb:144:in `upload'
/opt/chefdk/embedded/lib/ruby/gems/2.5.0/gems/inspec-3.6.6/lib/inspec/backend.rb:45:in `create': undefined method `unpack_train_credentials' for {:target=>"mock://"}:Hash (NoMethodError)
```

### 🌍 InSpec and Platform Version

3.6.6

### 🤔 Replication Case

Install version 3.6.6 and try uploading a controller to Chef Automate. 

### 💁 Possible Solutions

If you have already ideas how to solve the issue, add them here.

### 💻 Stacktrace

Please include the stacktrace output or link to a gist of it, if there is one.

**Comment by kmf** [comment]
I can confirm I had this as well.

**Comment by clintoncwolfe** [comment]
This was fixed in #3834, but we have not had a release since that was merged. Fix will be in the next release.

**Comment by clintoncwolfe** [comment]
Fixed in 3.7.1 release

**Comment by shyamdayals** [comment]
How can I download 3.7.1 ? "chef gem install inspec" seems to update it to 4.3.2, but "chef -v" still lists 3.6.6 version
